core/resource/service/resource_service.go

```diff
@@ -379,43 +436,44 @@ func (rs ResourceService) getResourcesToBatchUpdate(ctx context.Context, incomin
 	me := errors.NewMultiError("error in resources to batch update")
 
 	for _, incoming := range incomings {
-		if incoming.Status() != resource.StatusValidationSuccess {
-			continue
-		}
-
-		existing, ok := existingMappedByFullName[incoming.FullName()]
-		if !ok {
-			_ = incoming.MarkToCreate()
-			err := rs.repo.Create(ctx, incoming)
-			if err == nil {
-				toUpdateOnStore = append(toUpdateOnStore, incoming)
-			}
+		err := rs.identifyResourceToUpdateOnStore(ctx, incoming, existingMappedByFullName[incoming.FullName()])
+		if err != nil {
 			me.Append(err)
 			continue
 		}
-
-		if resource.StatusIsSuccess(existing.Status()) && incoming.Equal(existing) {
-			_ = incoming.MarkSkipped()
-			rs.logger.Warn("resource [%s] is skipped because it has no changes", existing.FullName())
-			continue
-		}
-
-		if resource.StatusForToCreate(existing.Status()) {
-			_ = incoming.MarkToCreate()
-		} else if resource.StatusForToUpdate(existing.Status()) {
-			_ = incoming.MarkToUpdate()
-		}
-
-		err := rs.repo.Update(ctx, incoming)
-		if err == nil {
+		if incoming.Status() == resource.StatusToCreate || incoming.Status() == resource.StatusToUpdate {
 			toUpdateOnStore = append(toUpdateOnStore, incoming)
 		}
-		me.Append(err)
 	}
 
 	return toUpdateOnStore, me.ToErr()
 }
 
+func (rs ResourceService) identifyResourceToUpdateOnStore(ctx context.Context, incoming, existing *resource.Resource) error {
+	if incoming.Status() != resource.StatusValidationSuccess {
+		return nil
+	}
+
+	if existing == nil {
+		_ = incoming.MarkToCreate()
+		return rs.repo.Create(ctx, incoming)
+	}
+
+	if resource.StatusIsSuccess(existing.Status()) && incoming.Equal(existing) {
+		_ = incoming.MarkSkipped()
+		rs.logger.Warn("resource [%s] is skipped because it has no changes", existing.FullName())
+		return nil
+	}
+
+	if resource.StatusForToCreate(existing.Status()) {
+		_ = incoming.MarkToCreate()
+	} else {
+		_ = incoming.MarkToUpdate()
```

Comment:
@sbchaos can you verify this change? as per my understanding, we should allow to update resources given any **_existing_** status. thus, making this change.

Reply:
Sure, this should be safe to do